packages/common/src/api/tan-query/useDeleteTrack.ts

```diff
@@ -0,0 +1,163 @@
+import { Id } from '@audius/sdk'
+import { useMutation, useQueryClient } from '@tanstack/react-query'
+import { useDispatch, useSelector } from 'react-redux'
+
+import { useAudiusQueryContext } from '~/audius-query'
+import { useAppContext } from '~/context/appContext'
+import { Name } from '~/models/Analytics'
+import { Feature } from '~/models/ErrorReporting'
+import { ID } from '~/models/Identifiers'
+import { Track } from '~/models/Track'
+import { UserMetadata } from '~/models/User'
+import { getWalletAddresses } from '~/store/account/selectors'
+import { deleteTrackSucceeded } from '~/store/cache/tracks/actions'
+
+import { getCurrentUserQueryKey } from './useCurrentUser'
+import { useCurrentUserId } from './useCurrentUserId'
+import { getTrackQueryKey } from './useTrack'
+import { useUser } from './useUser'
+import { primeTrackData } from './utils/primeTrackData'
+import { primeUserData } from './utils/primeUserData'
+
+type DeleteTrackArgs = {
+  trackId: ID
+  source?: string
+}
+
+type MutationContext = {
+  previousTrack: Track | undefined
+  previousUser: any | undefined
+}
+
+export const useDeleteTrack = () => {
+  const { audiusSdk, reportToSentry } = useAudiusQueryContext()
+  const queryClient = useQueryClient()
+  const dispatch = useDispatch()
+  const { data: currentUserId } = useCurrentUserId()
+  const { currentUser: currentUserWallet } = useSelector(getWalletAddresses)
+  const { data: currentUser } = useUser(currentUserId)
+  const {
+    analytics: { track: trackEvent }
+  } = useAppContext()
+
+  return useMutation({
+    mutationFn: async ({ trackId }: DeleteTrackArgs) => {
+      if (!currentUserId) throw new Error('User ID is required')
+      const sdk = await audiusSdk()
+
+      await sdk.tracks.deleteTrack({
+        trackId: Id.parse(trackId),
+        userId: Id.parse(currentUserId)
+      })
+
+      return { trackId }
+    },
+    onMutate: async ({ trackId, source }): Promise<MutationContext> => {
+      if (!currentUserId || !currentUser) {
+        throw new Error('User ID is required')
+      }
+
+      // Snapshot the previous values
+      const previousTrack = queryClient.getQueryData<Track>(
+        getTrackQueryKey(trackId)
+      )
+      if (!previousTrack) throw new Error('Track not found')
+
+      // Before deleting, check if the track is set as the artist pick & update if so
+      if (currentUser.artist_pick_track_id === trackId) {
```

Comment:
does this get handled on the backend ultimately?

Reply:
sry does what get handled?